src/tools/rbd_mirror/instance_watcher/Types.h

```diff
@@ -39,8 +39,110 @@ WRITE_CLASS_ENCODER(PeerImageId);
 
 typedef std::set<PeerImageId> PeerImageIds;
 
+enum NotifyOp {
+  NOTIFY_OP_IMAGE_ACQUIRE  = 0,
+  NOTIFY_OP_IMAGE_RELEASE  = 1,
+};
+
+struct ImageAcquirePayload {
+  static const NotifyOp NOTIFY_OP = NOTIFY_OP_IMAGE_ACQUIRE;
+
+  uint64_t request_id;
+  std::string global_image_id;
+  PeerImageIds peers;
```

Comment:
The source of information about peer is the PoolWatcher, which detects a mirroring image addition/removal in the pool. I thought pool watchers would be run only by the instance leader?

Now I think it is more correct to have only one peer in acquire/release notifications (not a list), because a message is going to be generated on a pool state change. Then actually we might not need `PeerImageId` type (and PeerImage). I added those mainly to be able to pass a list of peers (remote images). When it is always one peer, we can just pass mirror_uuid, image_id as arguments. And the newly introduced ImageReplayer::set_remote_images() should be removed and only add_remote_image()/remove_remote_image() are used.